evals/completion_fns/openai_inner_monologue.py

```diff
@@ -0,0 +1,67 @@
+"""
+Extending OpenAI Models with Chain-of-Thought Inner Monologue
+"""
+from evals import OpenAIChatCompletionFn, OpenAICompletionFn
+from evals.prompt.base import ChatCompletionPrompt, CompletionPrompt
+from evals.record import record_sampling
+
+
+class OpenAIInnerMonologueCompletionResult:
+    def __init__(self, response) -> None:
+        self.response = response
+
+    def get_completions(self) -> list[str]:
+        return [self.response.strip()]
+
+
+class OpenAIInnerMonologueCompletionFn:
+    def __init__(self, **kwargs) -> None:
+        self.cot_template = "\nBefore answering, I will reason in a step-by-step manner as to get the right answer, then conclude with the answer."
+        self.extract_answer_template = (
```

Comment:
These two variables could come from the constructor, which would make it much more customizable.